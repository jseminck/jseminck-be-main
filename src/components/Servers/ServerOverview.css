.server-container {
  margin-top: 100px;

  display: flex;
  flex-wrap: wrap;
  flex-direction: column;
  align-items: center;
  justify-content: center;
}

@keyframes spin {
  100%
  {
    transform:rotate(360deg);
  }
}