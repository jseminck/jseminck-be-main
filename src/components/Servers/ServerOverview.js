import React, {PropTypes} from 'react';
import {connect} from 'react-redux';
import {bindActionCreators} from 'redux';
import * as actions from './serverActions';
import '../../styles/server-overview.css';

export class ServerOverview extends React.Component {
  static propTypes = {
    actions: PropTypes.object.isRequired,
    serverOverview: PropTypes.object.isRequired
  };

  componentDidMount() {
    this.props.actions.onSetCurrentTime();

    this.props.serverOverview.servers.forEach(server => this.props.actions.onSetState(server));
  }

  render() {
    return (
      <div className="container">
        <div className="container-center">
          {this.props.serverOverview.servers.map((server, i) => {
            return (
              <div key={i} className="server-item">
                {this.getIcon(server.state)}
                <a href={`http://${server.name}.herokuapp.com`}>
                  {server.name}
                </a>
                &mdash;
                {server.heroku}
                {this.getDelay(server.delay)}
              </div>
            );
          })}
        </div>
      </div>
    );
  }

  getIcon(state) {
    if (state === "loading") {
      return <img src={require("./loading.png")} className="loading-icon" />;
    }
    if (state === "online") {
      return <img src={require("./online.svg")} className="online-icon" />;
    }
    if (state === "offline") {
      return <img src={require("./offline.svg")} className="offline-icon" />;
    }
  }

  getDelay(delay) {
    return delay > 0 ? `(${delay}s)` : "";
  }
}

function mapStateToProps(state) {
  return {serverOverview: state.serverOverview};
}

function mapDispatchToProps(dispatch) {
  return {actions: bindActionCreators(actions, dispatch)};
}

export default connect(mapStateToProps, mapDispatchToProps)(ServerOverview);