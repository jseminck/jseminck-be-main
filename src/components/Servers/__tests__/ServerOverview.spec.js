import React from "react";
import renderer from 'react-test-renderer';
import {ServerOverview} from "./../ServerOverview";

describe("ServerOverview", function() {
    it('renders correctly', () => {
        const app = renderer.create(
          <ServerOverview
              serverOverview={{
                servers: [{
                  link: 'myLink',
                  name: 'myName',
                  faIcon: 'myIcon'
                }]
              }}
              actions={{
                onSetCurrentTime: () => {},
                onSetState: () => {}
              }}
          />
        ).toJSON();

        expect(app).toMatchSnapshot();
    });
});