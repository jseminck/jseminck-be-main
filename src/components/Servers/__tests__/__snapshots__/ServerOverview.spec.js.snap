exports[`ServerOverview renders correctly 1`] = `
<div
  className="container">
  <div
    className="container">
    <div
      className="status"
      style={
        Object {
          "backgroundColor": undefined
        }
      } />
    <div
      className="icon">
      <i
        className="fa fa-3x fa-myIcon" />
    </div>
    <div
      className="name">
      <a
        href="http://myLink">
        myName
      </a>
    </div>
  </div>
</div>
`;
